### ISSUE TITLE
installation: stuck at welcome artix screen

### ISSUE THREAD

**Issue by ghost** [kyoz/mac-artix#1]
Waited for couple hours on that but can't see login input. Any idea? I am trying on a MacBookAir9,1

**Comment by kyoz** [comment]
Hi @ihsanturk , do you follow all the steps, especially the [install bootloader part](https://github.com/kyoz/mac-artix#install-the-bootloader) and [make artix dual boot](https://github.com/kyoz/mac-artix#make-artix-dual-bootable)

If so, when restart the mac, you can hold alt/option button and these 2 disk which you can choose: One is Artix and the other is OSX.

If you'll finish everything but still can't see anything, that's mean you have problem in install boot loader and make artix dual boot  with OSX, you have to boot to usb to mount again, install bootloader again. 

Pay attention of `boot.efi` file. It's may corrupt or not match the one you'v generated.

Please try again and let me know if i can help you anything else.

**Comment by kyoz** [comment]
Ah, i forgot, i'll you see the black screen with text running in it, you'v successfully install Artix, these is no login form, you have to login via that black screen, after login, you can install and tweak any thing you'd like, with this minimal install, the system just take 100MB from my Macbook and run smoothly, you then have to install what you'd like to use cause this is Artix. :D

**Comment by ghost** [comment]
No, I couldn't install Artix successfully. I am talking about logging in to the live iso. I think I could not explained the issue in detail. Here how I run into the problem

- ✅ I have made a partition from disk utility.
- ✅ dd'ed the Artix iso to my flash drive.
- ✅ Pressed "Option" key while booting and booted from flash drive (Artix Linux).
- ✅ I can see it is booting successfully, and when finally comes to "Welcome to Artix"
- ❌ I expect it asks login and password but it is freezing there so I can't log in to the live environment, thus I can't install anything. 

![IMG_9737](https://user-images.githubusercontent.com/24369769/123816510-0ca06e00-d900-11eb-9e19-5c7bee120452.jpeg)

I thought maybe the screen resolution does not match with the settings and login & password overflowing the screen, so I tried typing `artix<enter>artix<enter>` even if I don't see the input question, but nothing happens.

**Comment by kyoz** [comment]
During installation, did you install these:

```
basestrap /mnt base base-devel runit elogind-runit
```

Especially elogind-runit

**Comment by ghost** [comment]
I couldn't even begin installing the system since I can't login to live iso.

**Comment by kyoz** [comment]
I mean you have to [install the system](https://github.com/kyoz/mac-artix#install-base-system) before you install boot loader with basestrap, you have to [install linux firmware too](https://github.com/kyoz/mac-artix#install-kernel).

Then you have to install [these](https://github.com/kyoz/mac-artix#install-kernel) if you only have wireless.

I know it some how confuse, but it's harder than normal installation, all the steps i'v notes is required for the fresh artix to work and you can login and use it as normally.

**Comment by ghost** [comment]
But you are talking about the later steps. First of all, I can't even get in to live iso's command line!? How am I supposed to install the system. I can install if I can get to the live usb environment, but that doesn't load unfortunately. I know how to install the system afterwards, I did installed many distros (arch, void, etc.) before to a non-apple device, so the the installation process will not be a problem for me I guess. But first, I can't get into the installation command line yet. For clarity, I can only make progress to the  ["Boot it up" section](https://github.com/kyoz/mac-artix#boot-it-up) in the readme.md. So I can't even run `sudo pacman -Sy terminus-font` because of the main issue reason: system freezes on the "Welcome Artix linux" screen **on the live iso environment, not the installed Artix system**. 

When I boot from the iso USB for the first time **to install** Artix linux, I can't type a command due to system freeze at "Welcome to Artix" :/ I will then be able to run `basestrap` etc. whatever, but yeah... I am repeating myself

**Comment by ghost** [comment]
I tried nixos minimal iso, to see whether any other linux works on my mac, and it booted up but I couldn't use the internal keyboard. I searched a bit and found out that the [keyboard is not supported](https://discussions.apple.com/thread/251087440?answerId=252062188022#252062188022) in the linux kernel <v5.5 due to T2 security chip that newer macs have and drivers not supported. So I won't be able to install any linux distro until the linux kernel supports it.

Thank you for the responses anyway.

**Comment by ghost** [comment]
You can specify in the readme that the MacBooks that have T2 security chip does not yet supported.

**Comment by kyoz** [comment]
Thanks @ihsanturk

**Comment by kyoz** [comment]
Sorry @ihsanturk , i'v just read again and understand it, lol. I though you'll pass the setup. This issue is like you said. [New Macbooks with T2 Security](https://support.apple.com/en-us/HT208862) may not well support. But in the past i used to get similar problem with my boot usb. Turn out the dd and some tools to create usb boot may cause this problem. I remember i'v used a tool call Rufus on window (i borrow my friend Window Laptop) and it work without freeze. @@

BTW, can you send me the link where it say T2 Security Macbook not work. Thank you in advanced

**Comment by ghost** [comment]
Even if I write the ISO with a different tool, my keyboard will not work anyways. So not worth a try.

I already sent the link on that comment but here it is: https://discussions.apple.com/thread/251087440?answerId=252062188022#252062188022

**Comment by kyoz** [comment]
Oh nice, i'll add it to the docs. Hope there'll be an update or other thing to help you enjoy linux on your mac. Thank you very much :heart:

**Comment by ghost** [comment]
I decided to give a try to my external keyboard, I plugged it and it is working, so the issue right now is the overflowing tty screen size since I become sure by opening Vim, pressing <kbd>:</kbd> and can't seeing the ex mode line (it's out of the screen). I think this was the case on my artix installation attempt, it was not freezing on the welcome screen, I just couldn't see the login and password and even if I type `artix<enter>artix<enter>` without seeing them would not work because my keyboard was not working, so I thought that it was frozen.

Now, I am trying to find to set the screen size with kernel parameters but it seems that I can't (tried `vga=2560x1440@60`) but nothing changed. So, I can't see the very bottom and very right of the screen :/.

For the internal keyboard to work, [they say](https://github.com/Dunedan/mbp-2016-linux#keyboard--touchpad) I need to install BCE drivers and touchpad & keyboard patches. So I will also try those. Let's see...

**Comment by kyoz** [comment]
Great effort you are. You can tried to set the smaller font, i think i'll fully display with smaller font

**Comment by kyoz** [comment]
It may take time bro. kk. But it worth it
